### DISCUSSION TITLE
Set aside inputs that did not pass validation

### DISCUSSION THREAD

**Discussion by bentheiii** [pydantic/pydantic#4320] [Feature Request]
Hi, I have a use case where some inputs in a complex field might fail validation, in this case, I'd like the following to happen:

a. they should be excluded from the specific list
b. they should be stored elsewhere for further handling (logging, etc.)

my current solution is something like this:

```python

@dataclass
class ValidationFailure:
    err: ErrorList

class MightFail(Generic[T]):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v, field: ModelField):
        inner_f = field.sub_fields[0]
        valid_value, error = inner_f.validate(v, {}, loc='')
        if error:
            return Error(error)
        return valid_value

class MyModel(BaseModel):
    items: Dict[str, List[MightFail[Item]]]

    @root_validator(skip_on_failure=True)
    def clean_errors(cls, values):
        errs = []
        for key, item_list in values['items'].items():
            valid_items = []
            for item in item_list:
                if isinstance(item, Error):
                    errs.extend(item.err)
                else:
                    valid_items.append(item)
            item_list[:] = valid_items
        values['errs'] = errs
        return values
```

But maybe pydantic should support a more standardized solution

**Comment by samuelcolvin** [comment]
Should be possible using pydantic V2. You'll be about to set [`on_error`](https://github.com/samuelcolvin/pydantic-core/blob/da6b5687b2c55b09bb9d8163d849c672e31163c3/pydantic_core/_types.py#L126) on a field, but if you need to collect the failed values, you'd need to use a wrap validator and store things things in `context`.
